src/main/java/cpw/mods/fml/common/discovery/ModDiscoverer.java

```diff
@@ -82,10 +82,13 @@ else if (minecraftSources[i].isDirectory())
 
     }
 
-    public void findModDirMods(File modsDir)
+    public void findModDirMods(File[] modsDirs)
     {
-        File[] modList = FileListHelper.sortFileList(modsDir, null);
-        modList = FileListHelper.sortFileList(ObjectArrays.concat(modList, ModListHelper.additionalMods.values().toArray(new File[0]), File.class));
+        File[] modList = ModListHelper.additionalMods.values().toArray(new File[0]);
+        for (File dir : modsDirs)
+        {
+            modList = ObjectArrays.concat(modList, FileListHelper.sortFileList(dir, null), File.class);
```

Comment:
Seriously how hard is this findModDirMods(File.. dirs)
List<File> files = new ArrayList(adds);
for (dirs):
 if  dir exists files.add(dirs files)

sort files
load files